### PR TITLE
added cors allowed origin param to rest api section. also base apirou…

### LINKED ISSUE #27: Make Access Control Allowed Origin tunable
Currently, all the `settings.json` files have a field called `rpcAllowed`, but it is not used anywhere. I would assume that this was designed to take in some origins for which CORS would be allowed. 

At the same time, when you set `cors` to be `true`. `Access-Control-Allow-Origin` will be set to `*` in `ApiRoute.scala`.

Wouldn't it make more sense to control the `Access-Control-Allow-Origin` with the origins specified in `rpcAllowed`?

### PR DESCRIPTION
**Pull request by daron666**
…te has been reworked.

fixes #27

### CODE REVIEW

File: examples/src/main/scala/examples/commons/SimpleBoxTransaction.scala [hunk 1]
```diff
@@ -61,7 +61,7 @@ case class SimpleBoxTransaction(from: IndexedSeq[(PublicKey25519Proposition, Non
 
   override lazy val json: Json = Map(
     "id" -> Base58.encode(id).asJson,
-    "newBoxes" -> newBoxes.map(b => Base58.encode(b.id).asJson).asJson,
+    "newBoxes" -> newBoxes.map(b => Base58.encode(b.id).asJson).toSeq.asJson,
```
Comment (ceilican) [inline examples/src/main/scala/examples/commons/SimpleBoxTransaction.scala]: Why is this change needed?

File: src/main/resources/reference.conf [hunk 2]
```diff
@@ -27,6 +27,9 @@ scorex {
     # Enable/disable CORS support
     corsAllowed = false
 
+    # Allowed origin line for cors header value
+    corsAllowedOrigin = "*"
```
Comment (ceilican) [inline src/main/resources/reference.conf]: Instead of having `corsAllowed: Boolean` and `corsAllowedOrigin: String`, we could have just a single `corsAllowedOrigin: Option[String]`.  When it is `None`, cors is not allowed. When it is `Some(s)`, cors is allowed with origin `s`.

File: src/main/scala/scorex/core/api/http/ActorHelper.scala [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package scorex.core.api.http
+
+import akka.pattern.ask
+import akka.actor.ActorRef
+import akka.util.Timeout
+
+import scala.concurrent.Future
+import scala.reflect.ClassTag
+
+trait ActorHelper {
```
Comment (ceilican) [inline src/main/scala/scorex/core/api/http/ActorHelper.scala]: This trait is not specific to this package and could be useful elsewhere in scorex. I would suggest moving it to `scorex.core.utils`.

File: examples/src/main/scala/examples/commons/SimpleBoxTransaction.scala (hunk 1 above)
Comment (daron666) [reply examples/src/main/scala/examples/commons/SimpleBoxTransaction.scala]: I don't know why, but after updating circe's version this started to produce compile error.

File: src/main/scala/scorex/core/api/http/ActorHelper.scala (hunk 3 above)
Comment (daron666) [reply src/main/scala/scorex/core/api/http/ActorHelper.scala]: I think so too. So I will work according your comments and prepare pr in the nearest future.

**Comment by daron666** [comment]
created #170
